lisa/util/__init__.py

```diff
@@ -598,6 +601,58 @@ def is_valid_url(url: str, raise_error: bool = True) -> bool:
     return is_url
 
 
+def is_valid_source_code_package(
+    source_url: str,
+    expected_package_name_pattern: Pattern[str],
+    allowed_protocols: Optional[List[str]] = None,
+    expected_domains: Optional[List[str]] = None,
+) -> bool:
+    # avoid using a mutable default parameter
+    if not allowed_protocols:
+        allowed_protocols = [
+            "https",
+            "sftp",
+        ]
+    # first, check if it's a url.
+    if not is_valid_url(url=source_url, raise_error=False):
+        return False
```

Comment:
False is not easy to troubleshoot. You can add an argument called "raise_error" like other examples. If it's True, raise exceptions, otherwise log the exception and return False.